### PR TITLE
bsunanda:Run2-hcx106 Correct list of valid DetId's and add possibility of getting position of merged DetId

### PR DESCRIPTION
**Pull request by bsunanda**
Essential for Plan1 operation

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda the regular build error looks spurious, but the clang error seems real:
```
tmp/slc6_amd64_gcc530/MakeData/DirCache.mk:3: No such file exists: src/Geometry/HcalTowerAlgo/test/HcalDetIdTester.cc. Please fix src/Geometry/HcalTowerAlgo/test/BuildFile.
>> Local Products Rules ..... started
gmake: *** [src/Geometry/HcalTowerAlgo/test contains only BuildFile.xml without any sources. Better to remove BuildFile.xml too or do not tag this directory] Error 1
```
Can you fix it?

**Comment by bsunanda** [comment]
@smuzaffar This error is not due to this code. There is something else wrong in the environment for recent IB.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by smuzaffar** [comment]
please test

File: Geometry/HcalCommonData/interface/HcalDDDSimConstants.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ class HcalDDDSimConstants {
 					  const int zside) const;
   int                       getDepthEta16M(const int det) const;
   int                       getDepthEta29(const int phi, int zside, int i) const;
-  int                       getDepthEta29M(const int i) const;
+  int                       getDepthEta29M(const int i, const bool flag) const;
```
Comment (davidlange6) [inline Geometry/HcalCommonData/interface/HcalDDDSimConstants.h]: Hi @bsunanda - here and in a few other places in this PR - could you either document what "flag" means or rather pick a more self documenting name?

Comment (bsunanda) [inline Geometry/HcalCommonData/interface/HcalDDDSimConstants.h]: @davidlange6 Done that - this is for special RBX's as in Plan One - so I name it planOne

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch @davidlange6 Could you approve the request - I need to submit the next set of changes for Plan 1 once this is integrated

**Comment by davidlange6** [comment]
Will assume tests are ok given previous tests

**Pull request merged**